mb_modelbase/server/modelcache.py

```diff
@@ -0,0 +1,93 @@
+import abc
+from pymemcache.client import base as memcBase
+import redis
+
+class Cache():
+    def __init__(self, *args, **kwargs):
+        """Initialize Connections"""
+
+    @abc.abstractmethod
+    def get(self, key, default=None):
+        """ returns object for key or None"""
+
+    @abc.abstractmethod
+    def set(self, key, value):
+        """ Stores object with given key"""
+
+class DictCache(Cache):
+    def __init__(self, *args, **kwargs):
+        Cache.__init__(self,*args, **kwargs)
+        self._cache = {}
+
+    def set(self, key, value):
+        self._cache[key] = value
+
+    def get(self, key, default=None):
+        return self._cache.get(key, default)
+
+class MemcachedCache(Cache):
+    def __init__(self, *args, **kwargs):
+        Cache.__init__(*args, **kwargs)
+
+        self._expire_time = kwargs.get('memcached_expire_time', 0)
+
+        self._cache = memcBase.Client((
+            kwargs.get('hostname', 'localhost'),
+            kwargs.get('port', 11211),
+        ))
+
+    def set(self, key, value):
+        self._cache.set(
+            key,
+            value,
+            self._expire_time
+        )
+
+    def get(self, key, default=None):
+        return self._cache.get(key)
+
+class RedisCache(Cache):
+    def __init__(self, *args, **kwargs):
+        Cache.__init__(*args, **kwargs)
+
+        self._cache = redis.Redis(
+            host=kwargs.get('hostname', 'localhost'),
+            port=kwargs.get('port', 6379),
+            db=kwargs.get('redis_db', 0)
+        )
+
+    def set(self, key, value):
+        self._cache.set(
+            key,
+            value,
+        )
+
+    def get(self, key, default=None):
+        return self._cache.get(key)
+
+if __name__ == '__main__':
+
+
+    vals = range(1000)
+
+    dictCache = DictCache()
+    [dictCache.set(str(x), True) for x in vals]
+    res = [dictCache.get(str(x)) for x in vals]
+
+    # docker run --name memcached -d memcached:alpine
+    memCache = MemcachedCache({
+        'hostname': 'localhost',
+        'port': 11211
+    })
+
+    [memCache.set(str(x), str(x)) for x in vals]
+    res = [memCache.get(str(x)) for x in vals]
+
+    # docker run -p 6379:6379 --name lumen_redis_cache -d redis
+    redisCache = RedisCache({
+        'hostname': 'localhost',
+        'port': 6379
+    })
+
+    [redisCache.set( str(x), str(x)) for x in vals]
+    res = [redisCache.get(str(x)) for x in vals]
```

Comment:
What is this for? Should it be in a test case?

Reply:
The whole file is not needed anymore and has been removed.